File: ia/Class.js
//
// Class.js for super_zappy in /home/aracthor/programs/projects/hub/super_zappy/ia/scripts
// 
// Made by 
// Login   <aracthor@epitech.net>
// 
// Started on  Mon Nov  3 17:58:53 2014 
// Last Update Thu Nov 13 16:26:55 2014 
//

load("EAction.js");

function	Class(name)
{
    this.name = name;

    this.strength = 10;
    this.perception = 0;
    this.dexterity = 10;
    this.toughness = 10;
    this.intuition = 0;
    this.speed = 1;
    this.orientation = 1;
    this.crafting = 0;
    this.building = 0;
}

Class.prototype.presentationMessage = function()
{
    return (EAction.classDescription + ' ' +
	    this.name + ' ' +
	    this.strength + ' ' +
	    this.perception + ' ' +
	    this.dexterity + ' ' +
	    this.toughness + ' ' +
	    this.intuition + ' ' +
	    this.speed + ' ' +
	    this.orientation + ' ' +
	    this.crafting + ' ' +
	    this.building
	   );
}

File: ia/python/Class.py
#!/usr/bin/python3
## Class.py for super_zappy in /home/aracthor/programs/projects/hub/super_zappy/ia/python
## 
## Made by aracthor
## Login   <aracthor@epitech.net>
## 
## Started on  Wed Feb 25 09:13:41 2015 aracthor
## Last Update Wed Feb 25 09:20:14 2015 aracthor
##

from    Enums   import EAction

class   Class:
    def __init__(self, name):
        self.name = name

        self.strength = 10
        self.perception = 0
        self.dexterity = 10
        self.toughness = 10
        self.intuition = 0

        self.speed = 1
        self.orientation = 1
        self.crafting = 0
        self.building = 0
        self.shoot = 0


    def presentationMessage(self):
        return (EAction.CLASS_DESCRIPTION + ' ' +
                self.name + ' ' +
                str(self.strength) + ' ' +
                str(self.perception) + ' ' +
                str(self.dexterity) + ' ' +
                str(self.toughness) + ' ' +
                str(self.intuition) + ' ' +
                str(self.speed) + ' ' +
                str(self.orientation) + ' ' +
                str(self.crafting) + ' ' +
                str(self.building) + ' ' +
                str(self.shoot)
        )
